### DISCUSSION TITLE
Add more .env params for mysql collation

### DISCUSSION THREAD

**Discussion by tanthammar** [laravel/framework#40267] [Ideas] [answered]
Every time I create a new project I have to go to `config/database `and update the collation.
It would be nice if we could get two more env variables.

DB_CHARSET
DB_COLLATION

Then the `database.php `could change to:
```php
'mysql' => [
    //...
    'charset' => env('DB_CHARSET', 'utf8mb4'),
    'collation' => env('DB_COLLATION', 'utf8mb4_unicode_ci'),
    //...
]
```

**Comment by omarthegeek** [comment, marked as answer]
I believe that is possible now:
```
DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=laravel
DB_USERNAME=root
DB_PASSWORD=
DB_COLLATION=utf8mb4_unicode_ci
DB_CHARSET=utf8mb4
```

and you can confirm the setting via tinker:
```
php artisank tinker
env("DB_COLLATION")
env("DB_CHARSET")
```

**Reply by nicolus** [reply]
Your test from tinker does not confirm that the environment variable is actually used, only that it was set. To make sure it's actually being used you should run : 

```php
DB::getConfig('collation');
DB::getConfig('charset');
```

And it does work starting with Laravel 11 since it was recently added : https://github.com/laravel/laravel/commit/1ee78493898240a718d7d384811c3d2071924db4

If you updated your project from a previous version make sure you update your database.php config to use those environement variables.

**Reply by omarthegeek** [reply]
True. If you already had created the tables, changing the env variable won't do anything unless you did a fresh migration or you'll manually need to change the table database configuration directly from the database itself.
